src/test_dbl_linked_list.py

```diff
@@ -0,0 +1,312 @@
+"""Test dbl_linked_list data structures."""
+
+import pytest
+
```

Comment:
I really respect the quantity of your tests. However, there's a ton of repetition here and lots of code that must be maintained. You can get a lot more tests out of less code by using parametrized fixtures.